### PR TITLE
add k8s support for ehsm-kms

### PR DESCRIPTION
**Pull request by Liang1XZhao**
test: passed the unittest

Signed-off-by: Liang1XZhao <Liang1XZhao@intel.com>

### CODE REVIEW

File: ehsm_kms_service/kubernetes/ehsm-kms.yaml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        imagePullPolicy: IfNotPresent
+        readinessProbe:
+          httpGet:
+            port: couchdb
+            path: /
+          initialDelaySeconds: 5
+          periodSeconds: 10
+        ports: 
+        - containerPort: 5984
+          name: couchdb
```
Comment (syan10) [inline ehsm_kms_service/kubernetes/ehsm-kms.yaml:87]: seems you didn't change the name as i mentioned

File: ehsm_kms_service/kubernetes/ehsm-kms.yaml [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+      volumes:
+      - name: dev-enclave
+        hostPath:
+          path: /dev/sgx/enclave
+      - name: dev-aesmd
+        hostPath:
+          path: aesmd-socket
+      containers:
+      - name: ehsm-kms
+        image: ehsm_kms_service:latest
```
Comment (syan10) [inline ehsm_kms_service/kubernetes/ehsm-kms.yaml:139]: so currently, it assume the user need to load the docker image by himself?

File: ehsm_kms_service/kubernetes/ehsm-kms.yaml (hunk 1 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/ehsm-kms.yaml:87]: It will be change

File: ehsm_kms_service/kubernetes/ehsm-kms.yaml (hunk 2 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/ehsm-kms.yaml:139]: will be change to load local

File: ehsm_kms_service/kubernetes/ehsm-kms.yaml [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+      - name: dev-enclave
+        hostPath:
+          path: /dev/sgx/enclave
+      - name: dev-aesmd
+        hostPath:
+          path: aesmd-socket
+      containers:
+      - name: ehsm-kms
+        # You need to upload the image file of ehsm_kms_service to your docker server or load it to each work-node, and adjust intel/ehsm_kms_service:latest
+        image: intel/ehsm_kms_service:latest
```
Comment (syan10) [inline ehsm_kms_service/kubernetes/ehsm-kms.yaml:142]: You need to tag the ehsm-kms container image with this name on each worker node or change it to point to a docker hub to get the container image.

**Review by syan10 — COMMENTED** [review]
change the folder name to k8s to align the name and the commit title to
"add k8s support for ehsm-kms"

File: ehsm_kms_service/kubernetes/ehsm-kms.yaml (hunk 3 above)
Comment (Liang1XZhao) [reply ehsm_kms_service/kubernetes/ehsm-kms.yaml]: It will be change.

**Comment by Liang1XZhao** [comment]
> change the folder name to k8s to align the name and the commit title to "add k8s support for ehsm-kms"

It will be changes.

**Review by syan10 — APPROVED** [review]

**Review by yang8621 — APPROVED** [review]

**Pull request merged**